python/ray/train/mosaic/mosaic_trainer.py

```diff
@@ -0,0 +1,125 @@
+import inspect
+import os
+
+from ray.air import session
+from ray.train.constants import (
+    EVALUATION_DATASET_KEY,
+    TRAIN_DATASET_KEY,
+)
+
+from typing import TYPE_CHECKING, Callable, Dict, Optional
+from ray.air.checkpoint import Checkpoint
+from ray.air.config import DatasetConfig, RunConfig, ScalingConfig
+
+from ray.train.torch import TorchConfig, TorchTrainer
+from ray.train.trainer import GenDataset
+from ray.util import PublicAPI
+
+if TYPE_CHECKING:
+    from ray.data.preprocessor import Preprocessor
+
+import composer.trainer
```

Comment:
from composer.trainer import Trainer?

also 3rd party libraries should go into one block. And ray imports should go into another (separate) block.

Reply:
Reformatted!